### PR TITLE
Adapt Spring configuration without external tomcat.

### PR DESCRIPTION
**Pull request by tobiasake**
### Applicable Issues
Today RemRem require external Tomcat instance.
Configuration is not as simple as it could be and it is not possible to test application locally.

### Description of the Change
Adapt application so it is possible to execute it without external Tomcat.
Add possbility to configuret Spring from Java arguments and properties and take a application properties as input configuration. 
This simplifies a lot the configuration of RemRem in a container environment.

STATUS:
Works:
- Executing Spring from war file works.
    Executions examples:
    java -jar publish-service/target/publish-service-0.6.7.war --server.port=8037
    env "server.port=8037" java -jar publish-service/target/publish-service-0.6.7.war
- 'mvn spring-boot:run' works
  Executions examples:
  mvn spring-boot:run  -Dspring-boot.run.jvmArguments="-Dserver.port=8037"
-  RemRem-publish war file in TomCat installation with Java Spring properties works.
-  RemRem-publish war file in TomCat installation with catalina.home properties works.
   Provide the catalina.home property before start catalina.sh, e.g. 'catalina.home=/usr/local/tomcat' and property file should be placed at this path: /usr/local/tomcat/conf/config.properties

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.

Signed-off-by:

### CODE REVIEW

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java [hunk 1]
```diff
@@ -51,28 +62,51 @@
...
+                    String protocol= rabbitmqInstanceObject.get("mp").toString();
+                    log.info("Configuring RabbitMq instance for Eiffel message protocol: " + protocol);
+                  rabbitMqPropertiesMap.put(protocol, new RabbitMqProperties());
+                  rabbitMqPropertiesMap.get(protocol).setHost(rabbitmqInstanceObject.get("host").toString());
+                  rabbitMqPropertiesMap.get(protocol).setPort(Integer.parseInt(rabbitmqInstanceObject.get("port").toString()));
+                  rabbitMqPropertiesMap.get(protocol).setUsername(rabbitmqInstanceObject.get("username").toString());
+                  rabbitMqPropertiesMap.get(protocol).setPassword(rabbitmqInstanceObject.get("password").toString());
+                  rabbitMqPropertiesMap.get(protocol).setTlsVer(rabbitmqInstanceObject.get("tls").toString());
+                  rabbitMqPropertiesMap.get(protocol).setExchangeName(rabbitmqInstanceObject.get("exchangeName").toString());
+                  rabbitMqPropertiesMap.get(protocol).setDomainId(rabbitmqInstanceObject.get("domainId").toString());
```
[the remark this answers is not in the document]
Comment (tobiasake) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java]: Fixed.

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java [hunk 2]
```diff
@@ -51,28 +62,51 @@
...
+                        rabbitMqPropertiesMap.get(protocol).setPort(Integer.getInteger(entry.getValue().toString()));
+                    } else if (key.contains("rabbitmq.username")) {
+                        rabbitMqPropertiesMap.get(protocol).setUsername(entry.getValue().toString());
+                    } else if (key.contains("rabbitmq.password")) {
+                        rabbitMqPropertiesMap.get(protocol).setPassword(entry.getValue().toString());
+                    } else if (key.contains("rabbitmq.tls")) {
+                        rabbitMqPropertiesMap.get(protocol).setTlsVer(entry.getValue().toString());
+                    } else if (key.contains("rabbitmq.exchangeName")) {
+                        rabbitMqPropertiesMap.get(protocol).setExchangeName(entry.getValue().toString());
+                    } else if (key.contains("rabbitmq.domainId")) {
```
[the remark this answers is not in the document]
Comment (tobiasake) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java]: Not my changes. I think its appears only due these lines has moved down a bit in the file, and maybe some autoformatting from the IDE. This no code change from my side.

File: publish-service/pom.xml [hunk 3]
```diff
@@ -1,189 +1,223 @@
...
+		</dependency>
+	</dependencies>
+	<build>
+		<plugins>
+			<plugin>
+
+				<groupId>org.springframework.boot</groupId>
+				<artifactId>spring-boot-maven-plugin</artifactId>
+		        <configuration>    
+                    <mainClass>com.ericsson.eiffel.remrem.publish.App</mainClass>
```
[the remark this answers is not in the document]
Comment (tobiasake) [reply publish-service/pom.xml]: Ok, what formatting do we use for pom files?. This is Eclipse Maven Pom IDE default formatting settings.

File: publish-service/src/main/resources/application.properties [hunk 4]
```diff
@@ -0,0 +1,37 @@
+
+server.port=8986
+debug: false
+
```
[the remark this answers is not in the document]
Comment (tobiasake) [reply publish-service/src/main/resources/application.properties]: Fixed.

File: publish-service/src/main/resources/application.properties [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+#RabbitMq configurations
+
+# need to be updated according to the test env.
+#rabbitmq.host=mb101-eiffel010.lmera.ericsson.se
+# must exist
+#rabbitmq.exchange.name=eiffel.poc
+rabbitmq.instances.jsonlist=[{ "mp": "eiffelsemantics", "host": "127.0.0.1", "port": "5672", "username": "myuser", "password": "myuser", "tls": "", "exchangeName": "amq.direct", "domainId": "eiffelxxx" }, \
+{ "mp": "eiffel3", "host": "127.0.0.1", "port": "5682", "username": "myuser", "password": "myuser", "tls": "", "exchangeName": "amq.direct", "domainId": "eiffelxxx" }]
```
[the remark this answers is not in the document]
Comment (tobiasake) [reply publish-service/src/main/resources/application.properties]: Yes, it is a json format in yaml. How would you like the Json format should be in Yaml?

File: publish-service/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (xdurvak) [reply publish-service/pom.xml]: our eiffel code formatter works for pom.xml file also

[the remark this answers is not in the document]
Comment (tobiasake) [reply publish-service/pom.xml]: ok, tried to add that to eclipse settings, but seems to only works for Java and not pom. But find a xml tag setting that changed one tab to 4 spaces. I have pushed a new commit now that maybe fixed all indentations and tabs.

**Comment by tobiasake** [comment]
Version stepped in pom file and changelog md file updated.

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java [hunk 6]
```diff
@@ -51,28 +62,53 @@
...
             }
         }
-        for (Entry<String, Object> entry : map.entrySet())
-        {
-            String key = entry.getKey();
-            if (key.contains("rabbitmq")) {
-                String protocol = key.split("\\.")[0];
-                if (rabbitMqPropertiesMap.get(protocol) == null) {
-                    rabbitMqPropertiesMap.put(protocol, new RabbitMqProperties());
+        if (map.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (tobiasake) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java]: Fixed.

File: publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java [hunk 7]
```diff
@@ -19,40 +19,96 @@
...
     private Map<String, RabbitMqProperties> rabbitMqPropertiesMap = new HashMap<String, RabbitMqProperties>();
 
     /***
      * This method is used to give RabbitMq properties based on protocol
+     * 
      * @return protocol specific RabbitMq properties in map
      */
     public Map<String, RabbitMqProperties> getRabbitMqProperties() {
         Map<String, Object> map = new HashMap<String, Object>();
         String catalina_home = System.getProperty("catalina.home").replace('\\', '/');
```
[the remark this answers is not in the document]
Comment (tobiasake) [reply publish-common/src/main/java/com/ericsson/eiffel/remrem/publish/config/RabbitMqPropertiesConfig.java]: Fixed.

**Pull request merged**